qa-tests-backend/scripts/run-part-1.sh

```diff
@@ -108,6 +109,12 @@ test_part_1() {
     [[ ! -f FAIL ]] || die "Part 1 tests failed"
 }
 
+cleanup_part_1() {
+    info "Cleaning up after running part 1 of e2e tests"
+
+    cleanup_optional_e2e_components
+}
+
```

Comment:
@gavin-stackrox How we should handle that? here or in python teardown?

Reply:
@janisz @stehessel I'd suggest moving all the invocation and cleanup into `run-part-1.sh`. Having a `cleanup_optional_e2e_components()` in tests/e2e/lib.sh suggests to me that it will run whenever the `deploy_optional_e2e_components()` gets called, which is not the case.
It would be useful to have a generalized post cluster create / pre destroy hook but perhaps for a future PR.

Reply:
> It would be useful to have a generalized post cluster create / pre destroy hook but perhaps for a future PR.

I see that we have something like pre / post hooks in https://github.com/stackrox/stackrox/blob/master/.openshift-ci/pre_tests.py and https://github.com/stackrox/stackrox/blob/master/.openshift-ci/post_tests.py. Would this be a good place to call `setup_workload_identities` / `cleanup_workload_identities` from? Or should I leave it in `run-part-1.sh`?

Reply:
I'd put it all in `run-part-1.sh`.